diff --git a/daemon/org.freedesktop.impl.portal.Secret.service.in b/daemon/org.freedesktop.impl.portal.Secret.service.in
index bc4d921..e497a17 100644
--- a/daemon/org.freedesktop.impl.portal.Secret.service.in
+++ b/daemon/org.freedesktop.impl.portal.Secret.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.freedesktop.impl.portal.Secret
-Exec=@bindir@/gnome-keyring-daemon --start --foreground --components=secrets
+Exec=/usr/bin/dinitctl --quiet start gnome-keyring-daemon
diff --git a/daemon/org.freedesktop.secrets.service.in b/daemon/org.freedesktop.secrets.service.in
index 523675b..73197a6 100644
--- a/daemon/org.freedesktop.secrets.service.in
+++ b/daemon/org.freedesktop.secrets.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.freedesktop.secrets
-Exec=@bindir@/gnome-keyring-daemon --start --foreground --components=secrets
+Exec=/usr/bin/dinitctl --quiet start gnome-keyring-daemon
diff --git a/daemon/org.gnome.keyring.service.in b/daemon/org.gnome.keyring.service.in
index b782399..4136d56 100644
--- a/daemon/org.gnome.keyring.service.in
+++ b/daemon/org.gnome.keyring.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.gnome.keyring
-Exec=@bindir@/gnome-keyring-daemon --start --foreground --components=secrets
+Exec=/usr/bin/dinitctl --quiet start gnome-keyring-daemon
